omod/src/main/java/org/openmrs/module/appui/fragment/controller/SessionFragmentController.java

```diff
@@ -10,11 +11,18 @@
 import org.openmrs.ui.framework.fragment.action.ObjectResult;
 import org.springframework.web.bind.annotation.RequestParam;
 
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletResponse;
+
 public class SessionFragmentController {
 	
 	public FragmentActionResult setLocation(@RequestParam("locationId") Location location,
-	                   @SpringBean("locationService") LocationService locationService, UiSessionContext context) {
+											@SpringBean("locationService") LocationService locationService,
+											UiSessionContext context, ServletContext servletContext, HttpServletResponse httpResponse) {
 		context.setSessionLocation(location);
+		httpResponse.setHeader("Set-Cookie", AppUiConstants.COOKIE_NAME_LAST_SESSION_LOCATION + "=" + location.getLocationId()
```

Comment:
Since we are already using this in other reference application modules, just go ahead and change the openmrs version to 2.0.0 here https://github.com/openmrs/openmrs-module-appui/blob/master/pom.xml#L39
Then use setHttpOnly.
I think the current reference application already requires at least tomcat7 which should be happy with this. 😊